tla-io/src/main/scala/at/forsyte/apalache/tla/imp/passes/SanyParserPassImpl.scala

```diff
@@ -83,16 +83,17 @@ class SanyParserPassImpl @Inject() (
 
         // write parser output to specified destination, if requested
         options.get[String]("parser", "output").foreach { output =>
-          val file = new File(output)
+          val outfile = new File(output)
+          val outfileName = outfile.toString()
 
-          if (filename.toLowerCase.endsWith(".tla")) {
```

Comment:
This had been picking up the `filename` instead of the `outfile`, and the `filename` was the input file (artifact of shadowing and not having sufficiently careful test coverage.